app/controllers/api/v1/events_controller.rb

```diff
@@ -1,5 +1,40 @@
-class Api::V1::EventsController < ApplicationController
+class Api::V1::EventsController < ApiController
   def show
     @event = Event.find(params[:id])
   end
+
+  def create
+    @event = Event.new(event_params)
```

Comment:
You're duplicating a bunch here with the event_params.

Look into strong_parameters and if you really need a fully populated owner/user and not just and id.

If you do need to entire user/owner, you could `.merge(owner: user)` into the params hash.